mtags/src/main/scala-2/scala/meta/internal/pc/MetalsGlobal.scala

```diff
@@ -194,80 +195,126 @@ class MetalsGlobal(
-            }
+          if (history.isSymbolInScope(sym, pre)) {
+            TypeRef(NoPrefix, sym, args.map(arg => loop(arg, None)))
+          } else {
+            val ownerSymbol = pre.termSymbol
+            def hasConflictingMembersInScope =
+              history.lookupSymbol(sym.name).exists {
+                case _: LookupSucceeded => true
+                case _ => false
+              }
 
-          def shouldRenamePrefix =
-            !metalsConfig.isDefaultSymbolPrefixes || hasConflictingMembersInScope
-
-          history.config.get(ownerSymbol) match {
-            case Some(rename)
-                if shouldRenamePrefix && history.tryShortenName(
-                  ShortName(rename, ownerSymbol)
-                ) =>
-              TypeRef(
-                new PrettyType(rename.toString),
-                sym,
-                args.map(arg => loop(arg, None))
-              )
-            case _ =>
-              history.renames.get(sym) match {
-                case Some(rename)
-                    if history.nameResolvesToSymbol(rename, sym) =>
-                  TypeRef(
-                    NoPrefix,
-                    sym.newErrorSymbol(rename),
-                    args.map(arg => loop(arg, None))
-                  )
-                case _ =>
-                  if (
-                    sym.isAliasType &&
-                    (sym.isAbstract ||
-                      sym.overrides.lastOption.exists(_.isAbstract))
-                  ) {
-
-                    // Always dealias abstract type aliases but leave concrete aliases alone.
-                    // trait Generic { type Repr /* dealias */ }
-                    // type Catcher[T] = PartialFunction[Throwable, T] // no dealias
-                    loop(tpe.dealias, name)
-                  } else if (history.owners(pre.typeSymbol)) {
-                    if (history.nameResolvesToSymbol(sym.name, sym)) {
-                      TypeRef(NoPrefix, sym, args.map(arg => loop(arg, None)))
+            def shouldRenamePrefix =
+              !metalsConfig.isDefaultSymbolPrefixes || hasConflictingMembersInScope
+            history.config.get(ownerSymbol) match {
+              case Some(rename)
+                  if shouldRenamePrefix && history.tryShortenName(
+                    ShortName(rename, ownerSymbol)
+                  ) =>
+                TypeRef(
+                  new PrettyType(rename.toString),
+                  sym,
+                  args.map(arg => loop(arg, None))
+                )
+              case _ =>
+                history.renames.get(sym) match {
+                  case Some(rename)
+                      if history.nameResolvesToSymbol(rename, sym) =>
+                    TypeRef(
+                      NoPrefix,
+                      sym.newErrorSymbol(rename),
+                      args.map(arg => loop(arg, None))
+                    )
+                  case _ if history.isSymbolInScope(sym, pre) =>
```

Comment:
Is there any pattern to reach this case? I guess this pattern will be captured by https://github.com/scalacenter/scalafix/blob/69c0369ab49c41b0473ca68fe0a8ade396d3157e/scalafix-rules/src/main/scala/scala/meta/internal/pc/ScalafixGlobal.scala#L177 in the first place.

Reply:
Good catch! Removed it!